pages/upgrade-guide/4.0.md

```diff
@@ -0,0 +1,186 @@
+
+This change may affect existing code if certain scripts were expected to execute even when CSP headers were set. Either modify the CSP header to allow scripts from the same domain or remove the scripts.
+
+### Attributes
+
+#### Breaking change: `toggleClass(boolean | undefined)` signature removed
+
+This particular signature was deprecated in jQuery 3.0 and is now being removed in jQuery 4.0. After feedback and discussion with the community, it was determined that this signature caused unexpected behavior. For instance, if the argument to `toggleClass` was a variable that turned out to be `undefined` or `false`, it would result in all classes being removed. Also, while the `toggleClass( boolean )` signature was documented (and has been marked deprecated since 3.0), the `toggleClass( undefined )` signature was not.
+
+The recommendation for migration is to be explicit about which classes should be toggled.
+
+### Core
+
+#### Breaking change: Deprecated APIs removed
+
+The following APIs were deprecated in previous versions and have been removed in jQuery 4.0:
+
+* `jQuery.camelCase`
+	This function was never documented and was only used internally. It was deprecated in jQuery 3.0.
+
+* `jQuery.cssProps`
+	This object was used to map CSS properties to their vendor-prefixed versions, which is no longer necessary in jQuery 4.0's supported browsers.
+
+* `jQuery.isArray`
+* `jQuery.isFunction`
+* `jQuery.isNumeric`
+* `jQuery.isWindow`
+* `jQuery.fx.interval`
+* `jQuery.parseJSON`
+* `jQuery.nodeName`
+* `jQuery.now`
+* `jQuery.trim`
+* `jQuery.type`
+
+#### Breaking change: Remove undocumented Array methods
+
+jQuery 4.0 removes `push`, `sort`, and `splice` from the jQuery prototype. These methods were copies of the methods of the same names on the native Array prototype, but were never documented and were not intended for public use. Some plugins may have used them, so we are still marking this as a breaking change.
+
+#### Slim build breaking change: Remove `callbacks` and `deferred` modules
+
+The slim build of jQuery 4.0 no longer includes the `callbacks` and `deferred` modules. Use the full build if you need these modules, or use native Promise objects.
+
+#### Breaking change: Remove undocumented `root` parameter of `jQuery()`
+
+The `root` parameter of `jQuery()` was never documented and was only used internally. Also, it has not been necessary since jQuery 1.9 when `jQuery.sub()` was removed.
+
+#### Important fix: Add `exports` to package.json
+
+jQuery exports multiple builds that can be used in different environments. This includes the default build as both a universal module (UMD) and an ESM module, the slim build in UMD and ESM, and a factory build for running jQuery with DOM emulators like JSDOM. The `exports` field in `package.json` is used to specify which build is used in different environments. This change is not expected to affect existing code.
+
+### CSS
```

Comment:
From my summary:
```
* CSS: Remove the opacity CSS hook
    * 865469f5e60f55feb28469bb0a7526dd22f04b4e
    * The consequence is `.css( "opacity" )` will now return an empty string for detached elements in standard-compliant browsers and "1" in IE & the legacy Edge. That behavior is shared by most other CSS properties which we're not normalizing either.
```

Let's mention that now `.css( "opacity" )` on disconnected elements will return an empty string instead of `"1"` in modern browsers. I expect some code may depend on the `"1"` default.